extensions/wikia/ImageReview/ImageReviewEventsHooks.class.php

```diff
@@ -0,0 +1,83 @@
+<?php
+
+use \Wikia\Logger\WikiaLogger;
+use \Wikia\Tasks\Tasks\ImageReviewTask;
+
+class ImageReviewEventsHooks {
+	public static function onUploadComplete( UploadBase $form ) {
+		static::createAddTask( $form->getTitle() );
+
+		return true;
+	}
+
+	public static function onFileRevertComplete( Page $page ) {
+		static::createAddTask( $page->getTitle() );
+
+		return true;
+	}
+
+	public static function onArticleUndelete( Title $title, $created, $comment ) {
+		if ( static::isFileForReview( $title ) ) {
+			static::createAddTask( $title );
+		}
+
+		return true;
+	}
+
+	public static function onArticleDeleteComplete( Page $page, User $user, $reason, $articleId ) {
+		global $wgCityId;
+
+		$title = $page->getTitle();
+
+		if ( static::isFileForReview( $title ) ) {
```

Comment:
It's always a title as we have to have a valid page object coming into this method.